### PR TITLE
new resource: `azurerm_spring_cloud_new_relic_application_performance_monitoring`

### PR DESCRIPTION
**Pull request by ms-henglu**
```
=== RUN   TestAccSpringCloudNewRelicApplicationPerformanceMonitoring_basic
=== PAUSE TestAccSpringCloudNewRelicApplicationPerformanceMonitoring_basic
=== CONT  TestAccSpringCloudNewRelicApplicationPerformanceMonitoring_basic
=== RUN   TestAccSpringCloudNewRelicApplicationPerformanceMonitoring_requiresImport
=== PAUSE TestAccSpringCloudNewRelicApplicationPerformanceMonitoring_requiresImport
=== CONT  TestAccSpringCloudNewRelicApplicationPerformanceMonitoring_requiresImport
=== RUN   TestAccSpringCloudNewRelicApplicationPerformanceMonitoring_complete
=== PAUSE TestAccSpringCloudNewRelicApplicationPerformanceMonitoring_complete
=== CONT  TestAccSpringCloudNewRelicApplicationPerformanceMonitoring_complete
=== RUN   TestAccSpringCloudNewRelicApplicationPerformanceMonitoring_update
=== PAUSE TestAccSpringCloudNewRelicApplicationPerformanceMonitoring_update
=== CONT  TestAccSpringCloudNewRelicApplicationPerformanceMonitoring_update
--- PASS: TestAccSpringCloudNewRelicApplicationPerformanceMonitoring_requiresImport (542.21s)
--- PASS: TestAccSpringCloudNewRelicApplicationPerformanceMonitoring_basic (555.97s)
--- PASS: TestAccSpringCloudNewRelicApplicationPerformanceMonitoring_complete (637.54s)
--- PASS: TestAccSpringCloudNewRelicApplicationPerformanceMonitoring_update (693.14s)
PASS

Process finished with the exit code 0
```

### CODE REVIEW

File: internal/services/springcloud/spring_cloud_new_relic_application_performance_monitoring_resource.go
Comment (magodo) [inline internal/services/springcloud/spring_cloud_new_relic_application_performance_monitoring_resource.go:138]:
```suggestion
				return err
```

File: website/docs/r/spring_cloud_new_relic_application_performance_monitoring.html.markdown
Comment (magodo) [inline website/docs/r/spring_cloud_new_relic_application_performance_monitoring.html.markdown:46]: These boolean properties all have default values, so can we omit them in the example to make it simple?

File: website/docs/r/spring_cloud_new_relic_application_performance_monitoring.html.markdown
Comment (magodo) [inline website/docs/r/spring_cloud_new_relic_application_performance_monitoring.html.markdown:76]: Can we make this to be a list instead?

File: internal/services/springcloud/spring_cloud_new_relic_application_performance_monitoring_resource_test.go
Comment (magodo) [inline internal/services/springcloud/spring_cloud_new_relic_application_performance_monitoring_resource_test.go:161]: Can we change these values to be different than `basic`?

File: internal/services/springcloud/spring_cloud_new_relic_application_performance_monitoring_resource.go
Comment (magodo) [inline internal/services/springcloud/spring_cloud_new_relic_application_performance_monitoring_resource.go:117]:
```suggestion
```

File: internal/services/springcloud/spring_cloud_new_relic_application_performance_monitoring_resource_test.go
Comment (magodo) [inline internal/services/springcloud/spring_cloud_new_relic_application_performance_monitoring_resource_test.go:162]: Same here

File: internal/services/springcloud/spring_cloud_new_relic_application_performance_monitoring_resource_test.go
Comment (magodo) [inline internal/services/springcloud/spring_cloud_new_relic_application_performance_monitoring_resource_test.go:166]: Same here

File: internal/services/springcloud/spring_cloud_new_relic_application_performance_monitoring_resource_test.go
Comment (magodo) [inline internal/services/springcloud/spring_cloud_new_relic_application_performance_monitoring_resource_test.go:167]: Same here

**Review by magodo — CHANGES_REQUESTED** [review]
Thanks for this PR!
I have left some additional mostly minor comments that once addressed this should be good to merge 👍

**Comment by ms-henglu** [comment]
Thanks @magodo ! I've updated it as suggested.

File: internal/services/springcloud/spring_cloud_new_relic_application_performance_monitoring_resource_test.go
Comment (magodo) [inline internal/services/springcloud/spring_cloud_new_relic_application_performance_monitoring_resource_test.go:170]: The keys can be unquoted?

File: website/docs/r/spring_cloud_new_relic_application_performance_monitoring.html.markdown
Comment (magodo) [inline website/docs/r/spring_cloud_new_relic_application_performance_monitoring.html.markdown:42]: Same here

**Review by magodo — CHANGES_REQUESTED** [review]
Some minor comments, which LGTM once fixed.

File: internal/services/springcloud/spring_cloud_new_relic_application_performance_monitoring_resource_test.go
Comment (ms-henglu) [reply internal/services/springcloud/spring_cloud_new_relic_application_performance_monitoring_resource_test.go]: Thanks! I've fixed it.

**Comment by magodo** [comment]
LGTM!

**Review by katbyte — APPROVED** [review]
LGTM me too 🚡

**Pull request merged**
